packages/manager/modules/overthebox/src/overthebox/autoconfigure/multipath/autoconfigure-multipath.controller.js

```diff
@@ -0,0 +1,159 @@
+export default class OverTheBoxAutoconfigMultipath {
+  /* @ngInject */
+  constructor($http, $translate, OVERTHEBOX_MULTIPATH) {
+    this.$http = $http;
+    this.$translate = $translate;
+    this.OVERTHEBOX_MULTIPATH = OVERTHEBOX_MULTIPATH;
+  }
+
+  $onInit() {
+    this.displayAddInterfaceButton = true;
+    this.showAddInterface = false;
+
+    this.interface = {
+      gateway: '',
+      ifname: '',
+      interfaceName: '',
+      ipAddr: '',
+      isIPv6: false,
+      mtu: '',
+      netmask: '',
+      priority: '',
+      protocol: '',
+      multipath: '',
+      routingTable: '',
+      type: '',
+    };
+
+    return this.loadInterfaces();
+  }
+
+  loadInterfaces() {
+    return this.$http
+      .get(`/overTheBox/${this.serviceName}/configuration/network/interface`)
+      .then((response) => {
+        this.interfaces = response.data;
+        return response.data;
+      })
+      .catch((error) => {
+        // Display error message
+        this.errorMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_loading_failed',
+          {
+            errorMessage: error.data.message,
+          },
+        );
+      });
+  }
+
+  addInterface() {
+    this.interface.gateway = '';
+    this.interface.ifname = '';
+    this.interface.interfaceName = '';
+    this.interface.ipAddr = '';
+    this.interface.isIPv6 = false;
+    this.interface.mtu = '';
+    this.interface.netmask = '';
+    this.interface.priority = '';
+    this.interface.protocol = '';
+    this.interface.routingTable = '';
+    this.interface.type = '';
+    this.showAddInterface = true;
+    this.displayAddInterfaceButton = false;
+  }
+
+  createInterface() {
+    this.errorMessage = '';
+    this.successMessage = '';
+
+    const params = {
+      configuration: {
+        ifname: this.interface.ifname,
+        interfaceName: this.interface.interfaceName,
+      },
```

Comment:
```suggestion
    const params = {
      configuration: {
        ifname: this.interface.ifname,
        interfaceName: this.interface.interfaceName,
        gateway: this.interface.gateway
      },
```
to avoid that long list of `if` , it will result to the same

Reply:
I'm not sure that I have the same result with your suggestion, some are mandatory and others not, I don't want to send a blank value to the API if the value is not set.